buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -503,7 +504,9 @@ public ByteBuf getBytes(int index, ByteBuf dst, int length) {
 
     @Override
     public CharSequence getCharSequence(int index, int length, Charset charset) {
-        // TODO: We could optimize this for UTF8 and US_ASCII
+        if (charset == CharsetUtil.US_ASCII) {
```

Comment:
@njhill should we better use `equals(...)` ?

Reply:
Sure, I think they are cached but that would be safer, I will change it.